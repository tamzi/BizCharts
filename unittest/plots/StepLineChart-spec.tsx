import React from 'react';
import StepLineChart from '../../src/plots/StepLineChart';
import { render, cleanup } from '@testing-library/react';

const MOCK_DATA = [
  { year: '1991', value: 3 },
  { year: '1992', value: 4 },
  { year: '1993', value: 3.5 },
  { year: '1994', value: 5 },
  { year: '1995', value: 4.9 },
  { year: '1996', value: 6 },
  { year: '1997', value: 7 },
  { year: '1998', value: 9 },
  { year: '1999', value: 13 },
];

describe('Plots-StepLineChart', () => {
  test('兼容旧版StepLineChart', () => {
    let chart = null;
    render(<StepLineChart 
      data={MOCK_DATA}
      xField="year"
      yField="value"
      onGetG2Instance={
        (c) => chart = c
      }
      interactions={[
        {
          type: 'slider',
          cfg: {
            start: 0,
            end: 0.5,
          },
        },
      ]}
    />);
    expect(chart.options.slider).toEqual({
      start: 0,
      end: 0.5,
    });
    cleanup();
  })
  test('兼容旧版StepLineChart-point', () => {
    let chart = null;
    render(<StepLineChart 
      data={MOCK_DATA}
      xField="year"
      yField="value"
      point
      onGetG2Instance={
        (c) => chart = c
      }
    />);
    expect(chart.options.point).toEqual({});
    expect(chart.options).toMatchSnapshot();
    cleanup();
  })
  test('兼容旧版shape-shapeType', () => {
    let chart = null;
    render(<StepLineChart 
      data={MOCK_DATA}
      xField="year"
      yField="value"
      step="vhv"
      onGetG2Instance={
        (c) => chart = c
      }
    />);
    expect(chart.options.stepType).toBe('vhv');
    expect(chart.options).toMatchSnapshot();
    cleanup();
  })
})
