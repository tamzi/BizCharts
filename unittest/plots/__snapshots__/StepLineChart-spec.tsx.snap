// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Plots-StepLineChart 兼容旧版StepLineChart-point 1`] = `
Object {
  "animation": true,
  "appendPadding": Array [
    10,
    5,
    10,
    10,
  ],
  "autoFit": true,
  "data": Array [
    Object {
      "value": 3,
      "year": "1991",
    },
    Object {
      "value": 4,
      "year": "1992",
    },
    Object {
      "value": 3.5,
      "year": "1993",
    },
    Object {
      "value": 5,
      "year": "1994",
    },
    Object {
      "value": 4.9,
      "year": "1995",
    },
    Object {
      "value": 6,
      "year": "1996",
    },
    Object {
      "value": 7,
      "year": "1997",
    },
    Object {
      "value": 9,
      "year": "1998",
    },
    Object {
      "value": 13,
      "year": "1999",
    },
  ],
  "isStack": false,
  "legend": Object {
    "position": "top-left",
  },
  "point": Object {},
  "renderer": "canvas",
  "stepType": "hv",
  "tooltip": Object {
    "crosshairs": Object {
      "type": "x",
    },
    "shared": true,
    "showCrosshairs": true,
    "showMarkers": true,
  },
  "xAxis": Object {
    "label": Object {
      "autoHide": Object {
        "cfg": Object {
          "minGap": 6,
        },
        "type": "equidistance",
      },
      "autoRotate": false,
    },
    "nice": true,
  },
  "xField": "year",
  "yAxis": Object {
    "label": Object {
      "autoHide": true,
      "autoRotate": false,
    },
    "nice": true,
  },
  "yField": "value",
}
`;

exports[`Plots-StepLineChart 兼容旧版shape-shapeType 1`] = `
Object {
  "animation": true,
  "appendPadding": Array [
    10,
    5,
    10,
    10,
  ],
  "autoFit": true,
  "data": Array [
    Object {
      "value": 3,
      "year": "1991",
    },
    Object {
      "value": 4,
      "year": "1992",
    },
    Object {
      "value": 3.5,
      "year": "1993",
    },
    Object {
      "value": 5,
      "year": "1994",
    },
    Object {
      "value": 4.9,
      "year": "1995",
    },
    Object {
      "value": 6,
      "year": "1996",
    },
    Object {
      "value": 7,
      "year": "1997",
    },
    Object {
      "value": 9,
      "year": "1998",
    },
    Object {
      "value": 13,
      "year": "1999",
    },
  ],
  "isStack": false,
  "legend": Object {
    "position": "top-left",
  },
  "renderer": "canvas",
  "step": "vhv",
  "stepType": "vhv",
  "tooltip": Object {
    "crosshairs": Object {
      "type": "x",
    },
    "shared": true,
    "showCrosshairs": true,
    "showMarkers": true,
  },
  "xAxis": Object {
    "label": Object {
      "autoHide": Object {
        "cfg": Object {
          "minGap": 6,
        },
        "type": "equidistance",
      },
      "autoRotate": false,
    },
    "nice": true,
  },
  "xField": "year",
  "yAxis": Object {
    "label": Object {
      "autoHide": true,
      "autoRotate": false,
    },
    "nice": true,
  },
  "yField": "value",
}
`;
